iOS/Source/SunView.swift

```diff
@@ -1,33 +1,48 @@
 import UIKit
 import CoreLocation
 
-struct SunViewLocation {
-    let x: CGFloat
-    let y: CGFloat
-}
-
 class SunView: UIView {
     static let sunSize = CGFloat(18.0)
+    static let boundingWidth = UIScreen.main.bounds.width - 80
+    static let boundingHeight = CGFloat(108)
+
+    var firstTime = true
+
+    var animationInProgress = false {
+        didSet {
+            if self.animationInProgress == true {
+                self.currentTimeLabel.alpha = 0
+            } else {
+                UIView.animate(withDuration: 0.3) {
+                    self.currentTimeLabel.alpha = 1
+                }
+            }
+        }
+    }
 
     var sunPhase = SunPhase.predawn {
         didSet {
             self.moon.isHidden = true
```

Comment:
Tis should be better anyway... 🤔 If we distinguish all these 4 phases as `show-moon` and the others as `day` I should put that in the enum...